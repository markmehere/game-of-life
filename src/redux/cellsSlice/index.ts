import { cellsSlice } from "./cellsSlice";

export default cellsSlice.reducer;
