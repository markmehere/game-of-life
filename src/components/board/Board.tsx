import { useAppSelector } from "redux/hooks";
import styled from "styled-components";
import { selectBoard } from "redux/cellsSlice/cellsSlice";
import { Cell } from "components/board/Cell";

const Container = styled.div`
  border: 2px solid black;
  width: 30rem;
`;

const Row = styled.div``;

export function Board() {
  const board = useAppSelector(selectBoard);

  return (
    <Container>
      {board.map((row, r) => (
        <Row key={`row${r}`}>
          {row.map((cell, c) => (
            <Cell key={`cell${r},${c}`} value={cell} row={r} col={c} />
          ))}
        </Row>
      ))}
    </Container>
  );
}
