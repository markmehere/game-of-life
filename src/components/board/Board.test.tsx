import { screen as s, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "redux/store";
import { Board } from "./Board";

jest.mock("shared/constants", () => ({
  BOARD_COLS: 5,
  BOARD_ROWS: 3,
}));

describe("<Board />", () => {
  it("matches snapshot", async () => {
    const wrapper = render(
      <Provider store={store}>
        <Board />
      </Provider>
    );
    const buttons = await s.findAllByRole("button");
    expect(buttons).toHaveLength(15);
    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
