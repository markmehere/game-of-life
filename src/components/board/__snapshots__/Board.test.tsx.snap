// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> matches snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-gsDKAQ fCuNiF"
    >
      <div
        class="sc-dkPtRN"
      >
        <button
          aria-label="Cell 0,0"
          aria-pressed="false"
          class="sc-bdvvtL jrqIUZ"
        />
        <button
          aria-label="Cell 0,1"
          aria-pressed="false"
          class="sc-bdvvtL jrqIUZ"
        />
        <button
          aria-label="Cell 0,2"
          aria-pressed="false"
          class="sc-bdvvtL jrqIUZ"
        />
        <button
          aria-label="Cell 0,3"
          aria-pressed="false"
          class="sc-bdvvtL jrqIUZ"
        />
        <button
          aria-label="Cell 0,4"
          aria-pressed="false"
          class="sc-bdvvtL jrqIUZ"
        />
      </div>
      <div
        class="sc-dkPtRN"
      >
        <button
          aria-label="Cell 1,0"
          aria-pressed="false"
          class="sc-bdvvtL jrqIUZ"
        />
        <button
          aria-label="Cell 1,1"
          aria-pressed="false"
          class="sc-bdvvtL jrqIUZ"
        />
        <button
          aria-label="Cell 1,2"
          aria-pressed="false"
          class="sc-bdvvtL jrqIUZ"
        />
        <button
          aria-label="Cell 1,3"
          aria-pressed="false"
          class="sc-bdvvtL jrqIUZ"
        />
        <button
          aria-label="Cell 1,4"
          aria-pressed="false"
          class="sc-bdvvtL jrqIUZ"
        />
      </div>
      <div
        class="sc-dkPtRN"
      >
        <button
          aria-label="Cell 2,0"
          aria-pressed="false"
          class="sc-bdvvtL jrqIUZ"
        />
        <button
          aria-label="Cell 2,1"
          aria-pressed="false"
          class="sc-bdvvtL jrqIUZ"
        />
        <button
          aria-label="Cell 2,2"
          aria-pressed="false"
          class="sc-bdvvtL jrqIUZ"
        />
        <button
          aria-label="Cell 2,3"
          aria-pressed="false"
          class="sc-bdvvtL jrqIUZ"
        />
        <button
          aria-label="Cell 2,4"
          aria-pressed="false"
          class="sc-bdvvtL jrqIUZ"
        />
      </div>
    </div>
  </div>
</body>
`;
